src/components/plots/gridPlot.ts

```diff
@@ -43,28 +48,49 @@ export module Plot {
     }
 
     /**
-     * @param {string} attrToSet One of ["x", "y", "fill"]. If "fill" is used,
+     * @param {string} attrToSet One of ["x", "y", "x2", "y2", "fill"]. If "fill" is used,
      * the data should return a valid CSS color.
      */
     public project(attrToSet: string, accessor: any, scale?: Scale.AbstractScale<any, any>) {
+      attrToSet = attrToSet === "x1" ? "x" : attrToSet;
+      attrToSet = attrToSet === "y1" ? "y" : attrToSet;
       super.project(attrToSet, accessor, scale);
+
+      if (attrToSet === "x" && this._projections["x2"] === undefined) {
+        if (scale instanceof Scale.Category) {
+          super.project("x1", (d: any, i: number, u: any, m: Plot.PlotMetadata) => {
+            return scale.scale(this._projections["x"].accessor(d, i, u, m)) - scale.rangeBand() / 2;
+          });
+          super.project("x2", (d: any, i: number, u: any, m: Plot.PlotMetadata) => {
+            return scale.scale(this._projections["x"].accessor(d, i, u, m)) + scale.rangeBand() / 2;
+          });
+        } else if (scale instanceof Scale.AbstractQuantitative) {
+          super.project("x1", (d: any, i: number, u: any, m: Plot.PlotMetadata) => {
+            return scale.scale(this._projections["x"].accessor(d, i, u, m));
+          });
+        }
+      }
```

Comment:
What do you suggest to avoid the super.project calls? I think we had a discussion earlier about this and we decided that this was the way to go because x1/x2 projections had to be created, and they also had to be mapped correctly, and this was the most straightforward way to do it...?